### PR TITLE
refactor: modernize project

### PR DESCRIPTION
**Pull request by cobraz**
This pull request contains a lot of changes. As the title says, it modernizes the whole project setup. I don't think there is a file that isn't touched in this pull request.

- Introduces new `eslint` configurations (very improved, utilizes eslint, typescript-eslint, eslint-comments, prettier ...)
- Removes Lerna, replaced with yarn v2.
- Adds .gitattributes for generated files
- Renames file to follow kebab case (e.g. `desk-structure.js`)
- Improves prettier setup
- Removes styled-components and `@emotion/styled`
- Adds Github Actions workflow for pull requests (not only Netlify deploy preview). This checks build, lint and tests.
- Reorganize folders (mostly place components where they belong)

### CODE REVIEW

File: web/src/components/custom-link.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+//     transition: all 0.15s cubic-bezier(0.8, 0, 0.3, 1) 0s;
+//   }
+//   &&:hover:after {
+//     padding-left: 10px;
+//     padding-right: 4px;
+//     -webkit-transition: all 0.15s cubic-bezier(0.8, 0, 0.3, 1) 0s;
+//     transition: all 0.15s cubic-bezier(0.8, 0, 0.3, 1) 0s;
+//   }
+// `
+export const Link = GatsbyLink;
```
Comment (TanyaDolgopolova) [inline web/src/components/custom-link.tsx:27]: Why all code was removed? We have arrow animation in this file, will I need to rewrite this animation?

Comment (TanyaDolgopolova) [inline web/src/components/custom-link.tsx:27]: Why all code was removed? We have arrow animation in this file, will I need to rewrite this animation?

Comment (cobraz) [reply web/src/components/custom-link.tsx:27]: This file needs to be rewritten to not utilize `styled-components` or `emotion`

Comment (cobraz) [reply web/src/components/custom-link.tsx:27]: My thought was since you have a change on it, we'll implement those changes after this.

Comment (TanyaDolgopolova) [reply web/src/components/custom-link.tsx:27]: Okay, I thinking about fully delete this implementation and rewrite it with `react-icons` library

Comment (cobraz) [reply web/src/components/custom-link.tsx:27]: Sounds smart.

**Comment by cobraz** [comment]
> @cobraz how we will merge branches? At the start I think this branch need to be merged with develop branch first and then resolve conflicts with another

That is probably the best approach.

**Review by TanyaDolgopolova — APPROVED** [review]

**Pull request merged**